ultra/ultra/utils/curriculum/coordinator.py

```diff
@@ -28,23 +28,29 @@
 
 class Coordinator:
     def __init__(self, curriculum_dir, num_episodes):
+        """Initializes curriculum and sets up variables
+
+        Args:
+            curriculum_path (str): path to curriculum config file
+            num_episodes (int): total number of episodes
+        """
         CurriculumInfo.initialize(curriculum_dir)
 
-        self.mode = False
-        self.counter = cycle(tuple([i * 1 for i in range(self.get_num_of_grades())]))
-        self.eval_counter = cycle(
-            tuple([i * 1 for i in range(self.get_num_of_grades())])
-        )
-        self.grade_checkpoints = []
-        self.num_episodes = num_episodes
-        self.grade_counter = 0
-        self.episode_per_grade = 1
-        self.warmup_episodes = 1
+        self._train_counter = cycle(tuple([i * 1 for i in range(self.num_grades)]))
+        self._eval_counter = cycle(tuple([i * 1 for i in range(self.num_grades)]))
+        self._num_episodes = num_episodes
+        self._grade_counter = 0
+        self._episodes_per_grade = -1
```

Comment:
Just to clarify, is this incremented and used as an index somewhere (so that it starts at 0?)

Reply:
Yes, I will initialize that to zero and any other variables as well